### PR TITLE
fix misbehave of KLDivLoss

### PR DESCRIPTION
**Pull request by Neutron3529**
## Description ##
In the current version of KLDivLoss, the return value is not the same value calculated by SoftmaxCrossEntropyLoss, which is not documented. It may due to the incorrect settings which using mean rather than sum dealing with the return value.
I provide a fix of this setting, which will keep the return value of `KLDivLoss` and `SoftmaxCrossEntropyLoss` almost the same when `from_logits=False` and `sparse_label=False` are set to these functions seperately.
Now, the behave of KLDivLoss is exactly the same to what the document say.

to reproduce the misbehave in the current master branch:
```
import mxnet as mx
a=mx.nd.array([[-1,1],[1,-1]])
b=mx.nd.array([1,0]).one_hot(2)
TrueLoss=mx.gluon.loss.SoftmaxCrossEntropyLoss(sparse_label=False)
FalseLoss=mx.gluon.loss.KLDivLoss(from_logits=False)
c=TrueLoss(a,b)
d=FalseLoss(a,b)*a.shape[-1]
assert((c-d).abs().sum()==0 and a.shape[-1]==2)
```
### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at https://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [x] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [x] Now, the behave of KLDivLoss is exactly the same to what the document say.

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

**Comment by Neutron3529** [comment]
@mxnet-bot run ci [unix-cpu, unix-gpu, windows-gpu,  miscellaneous, windows-cpu]

**Comment by leezu** [comment]
Your PR breaks tests/python/gpu/test_gluon_gpu.py::test_sdml_loss

**Comment by Neutron3529** [comment]
To trigger all jobs: @mxnet-bot run ci [centos-gpu]

File: python/mxnet/gluon/loss.py [hunk 1]
```diff
@@ -1011,7 +1011,7 @@ def _compute_labels(self, F, batch_size):
         """
 
         # TODO: replace with mx.nd.eye(batch_size) with mxnet 1.2
-        gold = F.one_hot(F.arange(batch_size), batch_size)
+        gold = F.eye(batch_size)
```
Comment (szha) [inline python/mxnet/gluon/loss.py:1014]: remove outdated comment.

**Comment by Neutron3529** [comment]
> also, could you add a test case for the expected consistency?

I am so sorry that `I don't know where to add the test case`.

as what is documented for [SoftmaxCrossEntropyLoss](https://mxnet.apache.org/api/python/docs/api/gluon/loss/index.html#mxnet.gluon.loss.SoftmaxCrossEntropyLoss) and [KLDivLoss](https://mxnet.apache.org/api/python/docs/api/gluon/loss/index.html#mxnet.gluon.loss.KLDivLoss)

the return value of `SoftmaxCrossEntropyLoss()(pred,label)` and `KLDivLoss(from_logits=False)(pred,label.one_hot(number_of_classes))` should be the same.

i.e., the following assertion must be true
```
import mxnet as mx
from mxnet.gluon.loss import *
label=mx.nd.array([0,1,2,3,4,0,1,2,3,4])
pred=mx.nd.random.normal(shape=(10,5))
number_of_classes=pred.shape[1]
#using 1+...-1==0 to assure that only rounding errors would happened.
assert(1+(SoftmaxCrossEntropyLoss()(pred,label)-KLDivLoss(from_logits=False)(pred,label.one_hot(number_of_classes))).abs().mean()-1==0)
```

**Comment by Neutron3529** [comment]
@mxnet-bot run ci [unix-cpu]

**Comment by Neutron3529** [comment]
@szha Now, all the check is passed and the outdated comment is removed.
I know what the test case could be but don't know how to deploy it.
feel free to edit and merge my pull request :)
